### ISSUE TITLE
Trie should not be specific to subclasses of `AbstractString`

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/DataStructures.jl#220]
Tries work find for any collection of elements of a free monoid (I think I have that term right).
Not just strings,
but also things like arrays of Ints.
Technically arrays of Float64s but that causes other problems.

Wikipedia gets several examples on nonstrings.
I haven't have a close look at the code, but it seems like this is simply a matter of relaxing a type constraint.
Particularly since these are uncompressed dict based Tries.

**Comment by kmsquire** [comment]
@oxinabox, care to take a stab at implementing this?

**Comment by rofinn** [comment]
I could probably look into this. @oxinabox I'm not very familiar with the term "free monoid" and the wikipedia entry is a bit dense, but would that include `String`s or `Symbol`s?

**Comment by oxinabox** [comment]
Yeah, strings, but symbols (except if you convert them to strings).
We actually don't even need free moniods, thinking about it (because that requires a concat operation)

Basically a try can be indexed with 
anything that is a finite-length iterable of elements from a finite set really.

**Comment by rofinn** [comment]
@oxinabox Alright, I have a prototype of supporting both `AbstractString` and `Vector{T}`. https://github.com/JuliaCollections/Tries.jl/pull/2
